File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/Integration/Movie/MovieForm.cs
using Serenity.ComponentModel;

namespace UserControlForm.Integration.Forms;

[FormScript("Integration.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Name { get; set; }
    public int Type { get; set; }
    public string Description { get; set; }
}

File: UserControlForm/UserControlForm/UserControlForm.Web/Modules/ServerTypes/Integration/MovieForm.ts
import { StringEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface MovieForm {
    Name: StringEditor;
    Type: IntegerEditor;
    Description: StringEditor;
}

export class MovieForm extends PrefixedContext {
    static readonly formKey = 'Integration.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(MovieForm, [
                'Name', w0,
                'Type', w1,
                'Description', w0
            ]);
        }
    }
}